src/OpenIddict.Core/OpenIddictBuilder.cs

```diff
@@ -286,49 +284,7 @@ public class OpenIddictBuilder {
         /// </summary>
         /// <returns>The <see cref="OpenIddictBuilder"/>.</returns>
         public virtual OpenIddictBuilder AddEphemeralSigningKey() {
-            // Note: a 1024-bit key might be returned by RSA.Create() on .NET Desktop/Mono,
-            // where RSACryptoServiceProvider is still the default implementation and
-            // where custom implementations can be registered via CryptoConfig.
-            // To ensure the key size is always acceptable, replace it if necessary.
-            var algorithm = RSA.Create();
-
-            if (algorithm.KeySize < 2048) {
-                algorithm.KeySize = 2048;
-            }
-
-#if NET451
-            // Note: RSACng cannot be used as it's not available on Mono.
-            if (algorithm.KeySize < 2048 && algorithm is RSACryptoServiceProvider) {
-                algorithm.Dispose();
-                algorithm = new RSACryptoServiceProvider(2048);
-            }
-#endif
-
-            if (algorithm.KeySize < 2048) {
-                throw new InvalidOperationException("The ephemeral key generation failed.");
-            }
-
-            // Note: the RSA instance cannot be flowed as-is due to a bug in IdentityModel that disposes
-            // the underlying algorithm when it can be cast to RSACryptoServiceProvider. To work around
-            // this bug, the RSA public/private parameters are manually exported and re-imported when needed.
-            SecurityKey key;
-#if NET451
-            if (algorithm is RSACryptoServiceProvider) {
-                var parameters = algorithm.ExportParameters(includePrivateParameters: true);
-                key = new RsaSecurityKey(parameters);
-
-                // Dispose the algorithm instance.
-                algorithm.Dispose();
-            }
-
-            else {
-#endif
-                key = new RsaSecurityKey(algorithm);
-#if NET451
-            }
-#endif
-
-            return Configure(options => options.SigningCredentials.AddKey(key));
+            return Configure(options => options.SigningCredentials.AddEphemeralKey());
```

Comment:
We are now calling into [this](https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/blob/0df4b07e3b19ccf5bdb80e0eff44a07a4b217da7/src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerExtensions.cs#L227) ASOS helper right?